src/thor/bidirectional_astar.cc

```diff
@@ -145,70 +133,67 @@ bool BidirectionalAStar::ExpandForward(GraphReader& graphreader,
   }
 
   bool disable_uturn = false;
-  bool found_uturn = false;
   EdgeMetadata meta = EdgeMetadata::make(node, nodeinfo, tile, edgestatus_forward_);
-  EdgeMetadata uturn_meta = {};
+  EdgeMetadata uturn_meta{};
 
   // Expand from end node in forward direction.
-  for (uint32_t i = 0; i < nodeinfo->edge_count(); ++i, meta.increment_pointers()) {
+  for (uint32_t i = 0; i < nodeinfo->edge_count(); ++i, ++meta) {
 
     // Begin by checking if this is the opposing edge to pred.
     // If so, it means we are attempting a u-turn. In that case, lets wait with evaluating
     // this edge until last. If any other edges were emplaced, it means we should not
     // even try to evaluate a u-turn since u-turns should only happen for deadends
-    if (pred.opp_local_idx() == meta.edge->localedgeidx()) {
-      uturn_meta = meta;
-      found_uturn = true;
-      continue;
-    }
+    uturn_meta = pred.opp_local_idx() == meta.edge->localedgeidx() ? meta : uturn_meta;
 
-    disable_uturn = ExpandForwardInner(graphreader, pred, nodeinfo, pred_idx, meta, shortcuts, tile,
-                                       offset_time) ||
+    // Expand but only if this isnt the uturn, we'll try that later if nothing else works out
+    disable_uturn = (pred.opp_local_idx() != meta.edge->localedgeidx() &&
```

Comment:
since we dropped the `continue` above we can get the same effect by shortcircuiting the call to `ExpandForwardInner` here